st2tests/integration/mistral/base.py

```diff
@@ -42,8 +42,12 @@ def _wait_for_completion(self, execution, wait=300):
         for i in range(wait):
             eventlet.sleep(3)
             execution = self.st2client.liveactions.get_by_id(execution.id)
+
             if execution.status in ['succeeded', 'failed']:
-                break
+                if hasattr(execution, 'result') and 'tasks' in execution.result:
+                    if ([task for task in execution.result['tasks']
```

Comment:
Minor style thing - it would be a bit more readable if you moved the whole list comprehension out of the if statement condition:

```python
finished_tasks = [list comprehension]
if finished_tasks:
   break
```